Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -933,6 +933,80 @@ public AndConstraint<TAssertions> ContainInOrder(params object[] expected)
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        /// <summary>
+        /// Asserts the current collection does not contain the specified elements in the exact same order, not necessarily consecutive.
+        /// </summary>
+        /// <remarks>
+        /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+        /// </remarks>
+        /// <param name="unexpected">A <see cref="System.Array"/> with the unexpected elements.</param>
+        public AndConstraint<TAssertions> NotContainInOrder(params object[] unexpected)
+        {
+            return NotContainInOrder(unexpected, string.Empty);
+        }
+
+        /// <summary>
+        /// Asserts the current collection does not contain the specified elements in the exact same order, not necessarily consecutive.
+        /// </summary>
+        /// <remarks>
+        /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+        /// </remarks>
+        /// <param name="unexpected">An <see cref="IEnumerable"/> with the unexpected elements.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotContainInOrder(IEnumerable unexpected, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(unexpected, nameof(unexpected), "Cannot verify absence of ordered containment against a <null> collection.");
+
+            if (ReferenceEquals(Subject, null))
+            {
+                return new AndConstraint<TAssertions>((TAssertions)this);
+            }
+
+            IList<object> unexpectedItems = unexpected.ConvertOrCastToList<object>();
+            IList<object> actualItems = Subject.ConvertOrCastToList<object>();
+
+            if (unexpectedItems.Count > actualItems.Count)
+            {
+                return new AndConstraint<TAssertions>((TAssertions)this);
+            }
+
+            var actualItemsSkipped = 0;
+            for (int index = 0; index < unexpectedItems.Count && actualItems.Any(); index++)
+            {
+                object unexpectedItem = unexpectedItems[index];
+
+                actualItems = actualItems.SkipWhile(actualItem =>
+                {
+                    actualItemsSkipped++;
+                    return !actualItem.IsSameOrEqualTo(unexpectedItem);
```

Comment:
You're saying you're using `object/Equals` in the docs, but this method will first try to do a reference equality check, before considering the `Equals` implementation of the involved object.

Reply:
Yes, and the same goes for `ContainInOrder`. Should I mention this behavior in docs?

P.S. I have been noticing various inaccuracies in documentation of different assertions while writing the code for the latest PRs. Probably, he documentation verification and adjustment could be planned for the future versions of FluentAssertions?

Reply:
> Yes, and the same goes for ContainInOrder. Should I mention this behavior in docs?

No, it's totally fine. In the end, the result is the same, because if both the subject and the expectation are the _same_ object, then it doesn't matter if their `Equals` override does something non-standard.